sql/core.go

```diff
@@ -426,30 +426,26 @@ type IndexColumn struct {
 	Length int64
 }
 
-// IndexedTable represents a table that has one or more native indexes on its columns, and can use those indexes to
-// speed up execution of queries that reference those columns. Unlike DriverIndexableTable, IndexedTable doesn't need a
-// separate index driver to function.
-type IndexedTable interface {
-	IndexAddressableTable
-	// GetIndexes returns all indexes on this table.
-	GetIndexes(ctx *Context) ([]Index, error)
-}
-
-// IndexAddressable provides a Table that has its row iteration restricted to only the rows that match the given index
-// lookup.
+// IndexAddressable is a table that can be scanned through a primary index
 type IndexAddressable interface {
-	// WithIndexLookup returns a version of the table that will return only the rows specified by the given IndexLookup,
-	// which was in turn created by a call to Index.Get() for a set of keys for this table.
-	WithIndexLookup(IndexLookup) Table
+	// AsIndexedAccess returns a table that can perform scans constrained to an IndexLookup
```

Comment:
Probably should embed Table here too

Reply:
the tradeoff here is that non-table `IndexAddressable`s, like foreign key updaters, will have more unimplemented methods